Refactor the socket module API doc (#22528)
ballerina-platform/ballerina-lang: stdlib/socket/src/main/ballerina/src/socket/Module.md

> This module provides an implementation for connecting to a remote socket server or acts as a server for an incoming socket request. The module facilitates three types of endpoints called `socket:Client`, `socket:UdpClient`, and `socket:Listener`.

Reviewer: "provides an implementation for connecting to a remote socket server or acts as a server for an incoming socket request" 
^ The above doesn't sound right. Need to be rephrased properly.

Also, the "endpoint" terminology was dropped some time back. Again, needs to be rephrased in a better way

> #### TCP Client endpoints

Reviewer: Drop the use of the term endpoint.

Reviewer: Applies to other places as well, where clients and listeners are referred to as endpoints

> The `socket:Client` is used to connect to a socket server and interact with it. The client can only send the data to the server and the client's `callbackService` can retrieve the data from the server and do multiple requests/responses between the client and the server.

Reviewer: ```suggestion
The `socket:Client` is used to connect to a socket server and interact with it. The client can only send the data to the server and the client's call-back service can retrieve the data from the server and do multiple requests/responses between the client and the server.
```

A Client endpoint can be defined by providing the host and port as follows:

```ballerina

> socket:Client socketClient = new ({host: "localhost", port: 61598});

Reviewer 2: We need to mention the callback service as well here. template of the callback service.

Reviewer 2: And how the client can be used. following lines
```ballerina
    string msg = "Hello Ballerina\n";
    byte[] c1 = msg.toBytes();
    var writeResult = socketClient->write(c1);
```

```

#### UDP Client endpoints
The `socket:UdpClient` is used to interact with the remote UDP host and it can be defined as follows:
```ballerina

> socket:UdpClient socketClient = new;

Reviewer 2: We need to add code snippets and describe the usage of this client

Reviewer 2: like https://github.com/daneshk/ballerina/blob/nats-docs-updates/stdlib/messaging/nats/src/main/ballerina/src/nats/Module.md

Reviewer 2: It doesn't need to be complete code. only specify the important codes

> This module provides an implementation for connecting to a remote socket server or acts as a server for an incoming socket request. The module facilitates two types of endpoints called `socket:Client`, `socket:UdpClient` and `socket:Listener`.

Reviewer 3: ```suggestion
This module provides an implementation for connecting to a remote socket server or acts as a server for an incoming socket request. The module facilitates two types of endpoints called `socket:Client`, `socket:UdpClient` and `socket:Listener`.
```
Need to add the below sentence and link to any existing BBEs.
For information on the operations, which you can perform with this module, see the below **Functions**.

Author: I think, I already added. Do you mention https://github.com/ballerina-platform/ballerina-lang/pull/22528/files#diff-9b7e561f4f3218cb1989b22fbbd2a70eR29?

Reviewer 3: Reviewed.

> This module provides an implementation for connecting to a remote socket server or acts as a server for an incoming socket request. The module facilitates two types of endpoints called `socket:Client`, `socket:UdpClient` and `socket:Listener`.

Reviewer 4: ```suggestion
This module provides an implementation for connecting to a remote socket server or acts as a server for an incoming socket request. The module facilitates three types of endpoints called `socket:Client`, `socket:UdpClient` and `socket:Listener`.
```

#### TCP Client endpoints
The `socket:Client` is used to connect to a socket server and interact with it. The client can only send the data to the server and the client's `callbackService` can retrieve the data from the server and do multiple requests/responses between the client and the server.

> A Client endpoint can be defined by providing host and port as follows:

Reviewer 3: ```suggestion
A Client endpoint can be defined by providing the host and port as follows:
```

```ballerina
socket:Client socketClient = new ({host: "localhost", port: 61598});
```

#### UDP Client endpoints

> The `socket:UdpClient` is used to interact with the remote UDP host. 

Reviewer 3: ```suggestion
The `socket:UdpClient` is used to interact with the remote UDP host
```

> This endpoint can be defined as follows:

Reviewer 4: Shall we concatenate this sentence to above to get rid of `this endpoint` part?

> This endpoint can be defined as follows:

Reviewer 3: ```suggestion
and it can be defined as follows:
```

```ballerina
socket:UdpClient socketClient = new;
```

### Listener endpoints
The `socket:Listener` is used to listen to the incoming socket request. The `onConnect(socket:Caller)` resource function gets invoked when a new client is connected. The new client is represented using the `socket:Caller`.
The `onReadReady(socket:Caller)` resource gets invoked once the remote client sends some data.

> A `socket:Listener` endpoint can be defined as follows:

Reviewer 4: ```suggestion
A `socket:Listener` can be defined as follows:
```

```ballerina
listener socket:Listener server = new(61598);
```

For information on the operations, which you can perform with this module, see the below **Functions**. For examples on the usage of the operations, see the following.

>  * [Basic TCP Socket](https://ballerina.io/learn/by-example/tcp-socket-listener-client.html)

Reviewer 3: ```suggestion
 * [Basic TCP Socket Example](https://ballerina.io/learn/by-example/tcp-socket-listener-client.html)
```

>  * [Basic UDP Client Socket](https://ballerina.io/learn/by-example/udp-socket-client.html).

Reviewer 3: ```suggestion
 * [Basic UDP Client Socket Example](https://ballerina.io/learn/by-example/udp-socket-client.html).
```

> This module provides an implementation for sending/receiving messages to/from another application process (local or remote) for both connection-oriented and connectionless protocols.

Reviewer 4: ```suggestion
This module provides an implementation for sending/receiving messages to/from another application process (local or remote) for both connection-oriented and connectionless protocols. The module facilitates a TCP `Client` and a TCP `Listener`.
```

Author: Connection-oriented and connectionless protocols indicate TCP and UDP. So, I think, Don't need to update this.

#### TCP Client

The `socket:Client` is used to connect to a socket server and interact with it. The client can only send the data to the server and the client's call-back service can retrieve the data from the server and do multiple requests/responses between the client and the server.

> A Client can be defined by providing the host, port and callbackService as follows:

Reviewer 3: ```suggestion
A Client can be defined by providing the host, port, and callbackService as follows.
```
